### ISSUE TITLE
Getting erroneous 'no custom binaries provided' on unsupported platform

### ISSUE THREAD

**Issue by habovh** [prisma/prisma#8575]
### Bug description

> I'm opening this issue as suggested by @pantharshit00 in https://github.com/pantharshit00/prisma-rpi-builds/issues/2#issuecomment-892599496

I'm on a Raspberry Pi (armv7) and using custom built binaries because of lack of official support for this platform.

Today I updated from 2.18.0 to 2.28.0 and I realised that in addition to the custom binaries I now also needed to patch Prisma's code inside node modules.

Lack of patch would result in this output when running `NODE_ENV=production yarn prisma generate`:

```
habovh@raspberrypi2:~/prisma-project $ NODE_ENV=production DEBUG="*" yarn prisma generate
yarn run v1.22.5
$ /home/habovh/prisma-project/node_modules/.bin/prisma generate
  prisma:loadEnv project root found at /home/habovh/prisma-project/package.json +0ms
  prisma:tryLoadEnv Environment variables not found at null +0ms
  prisma:tryLoadEnv Environment variables loaded from ./prisma/.env +3ms
[dotenv][DEBUG] did not match key and value when parsing line 5: 
[dotenv][DEBUG] did not match key and value when parsing line 7: 
Environment variables loaded from prisma/.env
  prisma:engines using Node API: false +0ms
  prisma:engines binaries to download query-engine, migration-engine, introspection-engine, prisma-fmt +4ms
  prisma:download Using env var PRISMA_QUERY_ENGINE_BINARY for binary query-engine, which points to /home/habovh/prisma-engines-dl/query-engine +0ms
  prisma:download Using env var PRISMA_MIGRATION_ENGINE_BINARY for binary migration-engine, which points to /home/habovh/prisma-engines-dl/migration-engine +2ms
  prisma:download Using env var PRISMA_INTROSPECTION_ENGINE_BINARY for binary introspection-engine, which points to /home/habovh/prisma-engines-dl/introspection-engine +2ms
  prisma:download Using env var PRISMA_FMT_BINARY for binary prisma-fmt, which points to /home/habovh/prisma-engines-dl/prisma-fmt +1ms
Prisma schema loaded from prisma/schema.prisma
  prisma:download Using env var PRISMA_QUERY_ENGINE_BINARY for binary query-engine, which points to /home/habovh/prisma-engines-dl/query-engine +0ms
  prisma:getConfig Using Query Engine Binary at: /home/habovh/prisma-engines-dl/query-engine +0ms
  prisma:getDMMF Using Query Engine Binary at: /home/habovh/prisma-engines-dl/query-engine +0ms
Error: Error: Unknown binary target arm-openssl-1.1.x in generator client.
Possible binaryTargets: darwin, darwin-arm64, debian-openssl-1.0.x, debian-openssl-1.1.x, rhel-openssl-1.0.x, rhel-openssl-1.1.x, linux-arm-openssl-1.1.x, linux-arm-openssl-1.0.x, linux-musl, linux-nixos, windows, freebsd11, freebsd12, openbsd, netbsd, arm, native
    at validateGenerators (/home/habovh/prisma-project/node_modules/prisma/build/index.js:34665:19)
    at async Object.getGenerators (/home/habovh/prisma-project/node_modules/prisma/build/index.js:34419:5)
    at async _Generate.parse (/home/habovh/prisma-project/node_modules/prisma/build/index.js:65886:22)
    at async main (/home/habovh/prisma-project/node_modules/prisma/build/index.js:113324:18)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Here's the patch that I'm using (applied with `patch-package` as postinstall):

> prisma+2.28.0.patch
```patch
diff --git a/node_modules/prisma/build/index.js b/node_modules/prisma/build/index.js
index 33a01a1..f2193a0 100755
--- a/node_modules/prisma/build/index.js
+++ b/node_modules/prisma/build/index.js
@@ -12146,6 +12146,7 @@ var require_platforms = __commonJS((exports2) => {
     "debian-openssl-1.1.x",
     "rhel-openssl-1.0.x",
     "rhel-openssl-1.1.x",
+    "arm-openssl-1.1.x",
     "linux-arm-openssl-1.1.x",
     "linux-arm-openssl-1.0.x",
     "linux-musl",
diff --git a/node_modules/prisma/prisma-client/generator-build/index.js b/node_modules/prisma/prisma-client/generator-build/index.js
index 521d129..b348f4a 100644
--- a/node_modules/prisma/prisma-client/generator-build/index.js
+++ b/node_modules/prisma/prisma-client/generator-build/index.js
@@ -12595,6 +12595,7 @@ var require_platforms2 = __commonJS2((exports2) => {
     "debian-openssl-1.1.x",
     "rhel-openssl-1.0.x",
     "rhel-openssl-1.1.x",
+    "arm-openssl-1.1.x",
     "linux-arm-openssl-1.1.x",
     "linux-arm-openssl-1.0.x",
     "linux-musl",
@@ -28290,6 +28291,7 @@ var require_dist16 = __commonJS2((exports, module) => {
       "debian-openssl-1.1.x",
       "rhel-openssl-1.0.x",
       "rhel-openssl-1.1.x",
+      "arm-openssl-1.1.x",
       "linux-arm-openssl-1.1.x",
       "linux-arm-openssl-1.0.x",
       "linux-musl",
```

### How to reproduce

1. Build or download binaries
2. Setup prisma's env to use custom binaries
3. Run `NODE_ENV=production yarn prisma generate`

### Expected behavior

It is expected that Prisma uses custom binaries and does not complain about an unsupported platform.

### Prisma information

Not relevant to this issue

### Environment & setup

- OS: Raspbian armv7l
- Database: not relevant to this issue
- Node.js version: v16.6.1

### Prisma Version

```
Environment variables loaded from prisma/.env
prisma               : 2.28.0
@prisma/client       : 2.28.0
Current platform     : arm-openssl-1.1.x
Query Engine         : query-engine 89facabd0366f63911d089156a7a70125bfbcd27 (at ../../prisma-engines-dl/query-engine, resolved by PRISMA_QUERY_ENGINE_BINARY)
Migration Engine     : migration-engine-cli 89facabd0366f63911d089156a7a70125bfbcd27 (at ../../prisma-engines-dl/migration-engine, resolved by PRISMA_MIGRATION_ENGINE_BINARY)
Introspection Engine : introspection-core 89facabd0366f63911d089156a7a70125bfbcd27 (at ../../prisma-engines-dl/introspection-engine, resolved by PRISMA_INTROSPECTION_ENGINE_BINARY)
Format Binary        : prisma-fmt 89facabd0366f63911d089156a7a70125bfbcd27 (at ../../prisma-engines-dl/prisma-fmt, resolved by PRISMA_FMT_BINARY)
Default Engines Hash : 89facabd0366f63911d089156a7a70125bfbcd27
Studio               : 0.417.0
```

**Comment by pantharshit00** [comment]
Hmm, problem here is we named our 64 bit arm platform as `linux-arm-openssl-1.x.x` instead of `linux-arm64-openssl-1.x.x`: https://github.com/prisma/engines-wrapper/blob/e0f2a284b4f6bc8ca225d6e2cf044e7d86e671b3/packages/get-platform/src/getPlatform.ts#L201-L203

Either we change `linux-arm` to `linux-arm64` which I think we can't really do now, or add `linux-arm32` for 32 bit arm as known target. We are not going to provide precompiled binaries for it anyways.

**Comment by pantharshit00** [comment]
Made: https://github.com/prisma/engines-wrapper/pull/125

I will wait for others to comment.

**Comment by pantharshit00** [comment]
Now this will tackled via https://github.com/prisma/prisma/issues/8653.

**Comment by pantharshit00** [comment]
Hey @habovh 

I fixed this in `3.1.0-dev.12`. You can try that release or wait for a stable release which is planned for next Tuesday(21 Sept)
